src/app.js

```diff
@@ -243,22 +243,28 @@ var run = function() {
 	$vmToggle.on('change', executionContextChange );
 	$web3Toggle.on('change', executionContextChange );
 	$web3endpoint.on('change', function() {
-		var endpoint = $web3endpoint.val();
-		if (endpoint == 'ipc')
-			web3.setProvider(new web3.providers.IpcProvider());
-		else
-			web3.setProvider(new web3.providers.HttpProvider(endpoint));
-		compile();
+		setProviderFromEndpoint();
+		if (executionContext == 'web3') compile();
```

Comment:
I'm making two assumptions here:
* the only possible values for `executionContext` are `web3` and `vm`
* the value of `endpoint` is meaningless while in `vm` context